src/ray/raylet_client/raylet_client.h

```diff
@@ -256,45 +256,11 @@ class RayletClientInterface : public PinObjectsInterface,
 
 namespace raylet {
 
-class RayletConnection {
- public:
-  /// Connect to the raylet.
-  ///
-  /// \param raylet_socket The name of the socket to use to connect to the raylet.
-  /// \param worker_id A unique ID to represent the worker.
-  /// \param is_worker Whether this client is a worker. If it is a worker, an
-  ///        additional message will be sent to register as one.
-  /// \param job_id The ID of the driver. This is non-nil if the client is a
-  ///        driver.
-  /// \return The connection information.
-  RayletConnection(instrumented_io_context &io_service,
-                   const std::string &raylet_socket,
-                   int num_retries,
-                   int64_t timeout);
-
-  ray::Status WriteMessage(MessageType type,
-                           flatbuffers::FlatBufferBuilder *fbb = nullptr);
-
-  ray::Status AtomicRequestReply(MessageType request_type,
-                                 MessageType reply_type,
-                                 std::vector<uint8_t> *reply_message,
-                                 flatbuffers::FlatBufferBuilder *fbb = nullptr);
-
- private:
-  /// Shutdown the raylet if the local connection is disconnected.
-  void ShutdownIfLocalRayletDisconnected(const Status &status);
-  /// The connection to raylet.
-  std::shared_ptr<ServerConnection> conn_;
-  /// A mutex to protect stateful operations of the raylet client.
-  std::mutex mutex_;
-  /// A mutex to protect write operations of the raylet client.
-  std::mutex write_mutex_;
-};
-
 class RayletClient : public RayletClientInterface {
```

Comment:
can we have a comment here about what's difference of RayletClient vs RayletConnection ? Something like: RayletConnection is a simple connection, whereas RayletClient sends Register (and Announce) RPCs in ctor to mark this core worker as ready (IIUC)

Reply:
I added doc for raylet connection.

Reply:
> RayletClient sends Register (and Announce) RPCs in ctor to mark this core worker as ready (IIUC)

raylet client is way more complicated than that, checkout the interface
https://github.com/ray-project/ray/blob/1b07eaf9163853bfa429f86328443db30aee903b/src/ray/raylet_client/raylet_client.h#L222-L227

Tried to add a simple wording.